### PR TITLE
Update trilinos, hypre and fmt.

### PR DESCRIPTION
**Pull request by CusiniM**
- Trilinos: 15.1.1
- Hypre: hypre-v2.31.0-26-gf6cfb0355 
- fmt 11.0.1

### CODE REVIEW

**Review by rrsettgast — APPROVED** [review]

**Comment by CusiniM** [comment]
I could not find a solution for any of the centos based images. I tried [this](https://serverfault.com/questions/1161816/mirrorlist-centos-org-no-longer-resolve) but could not really get it to fully work. Only works for the base repo. I have added a rockylinux job with gcc to replace the centOs job but I had to remove Pecan's jobs. I have also removed the `Pangea2` one since the system was retired. 

@rrsettgast please add the folder for the new job to the bucket.

**Comment by victorapm** [comment]
Given the updates in this PR, should we incorporate #272 here as well?

File: CMakeLists.txt [hunk 1]
```diff
@@ -378,7 +378,6 @@ ExternalProject_Add( chai
                      URL ${CHAI_URL}
                      INSTALL_DIR ${CHAI_DIR}
                      DEPENDS raja
-                     PATCH_COMMAND patch -p0 < ${TPL_MIRROR_DIR}/umpire_fcompare-debug-second.patch
                      BUILD_COMMAND ${TPL_BUILD_COMMAND}
```
Comment (CusiniM) [inline CMakeLists.txt:381]:
```suggestion
                     PATCH_COMMAND patch -p0 < ${TPL_MIRROR_DIR}/umpire_fcompare-debug-second.patch
                     BUILD_COMMAND ${TPL_BUILD_COMMAND}
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -370,15 +370,14 @@ list(APPEND build_list raja )
 # CHAI
 ################################
 set(CHAI_DIR "${CMAKE_INSTALL_PREFIX}/chai")
-set(CHAI_URL "${TPL_MIRROR_DIR}/chai-2023.06.0.tar.gz")
+set(CHAI_URL "${TPL_MIRROR_DIR}/chai-2024.02.2.tar.gz")
```
Comment (CusiniM) [inline CMakeLists.txt:373]:
```suggestion
set(CHAI_URL "${TPL_MIRROR_DIR}/chai-2023.06.0.tar.gz")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -332,7 +332,7 @@ list(APPEND build_list silo )
 
 
 set(RAJA_DIR "${CMAKE_INSTALL_PREFIX}/raja")
-set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-v2023.06.1.tar.gz")
+set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-v2024.02.2.tar.gz")
```
Comment (CusiniM) [inline CMakeLists.txt:335]:
```suggestion
set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-v2023.06.1.tar.gz")
```

**Pull request merged**